src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionData.cs

```diff
@@ -21,6 +21,11 @@ public class TriggeredFunctionData
         /// </summary>
         public object TriggerValue { get; set; }
 
+        /// <summary>
+        /// The ID for the message being processed. Not applicable for all trigger types.
+        /// </summary>
+        public string MessageId { get; set; }
```

Comment:
Let's not plumb this message ID through all these core types, just to get a log statement. Instead, just add a new log statement at the correct place in the listener for triggers that need it. E.g. for Queues [here](https://github.com/Azure/azure-webjobs-sdk/blob/554b7ba922be3a4e1f380034dc0c62d4efb2aa79/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueListener.cs#L263)

Reply:
We need to make sure that we can correlate the invocationId with the message id... I'm not sure we have an invocation id at that stage, do we?

Reply:
Right. Not until the listener calls into the executor is the function instance ID known. How about we generalize this then, and add a new "string TriggerMessage" property here, which allows a listener to optionally specify a log message that will be written by the executor as part of the invocation. Then in FunctionExecutor IIF there is a value for this message it gets logged, as a new log line (not part of the executing/executed messages), and it gets logged under the same FunctionInvocationId.

Reply:
That sounds good to me. How do we want to handle something like batched EH? List the count plus all the ids in the batch?

Reply:
Details need to be a very short summary, so you definitely can't have all the Event IDs in a batch. Here are some suggested details:

- Queues: "MessageId: X, DequeueCount: Y, InsertionTime: Z"
- ServiceBus: "MessageId: X, DeliveryCount: Y, EnqueuedTime: Z"
- EventHub
    - Batched: "BatchCount: X"
    - Single: "SequenceNumber: X, Offset: Y, EnqueuedTime: Z"